File: Longest Common Prefix - Leetcode 14/Longest Common Prefix - Leetcode 14.js
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) return "";
    
    let minLength = Infinity;
    for (const s of strs) {
        minLength = Math.min(minLength, s.length);
    }
    
    let i = 0;
    while (i < minLength) {
        for (const s of strs) {
            if (s[i] !== strs[0][i]) {
                return strs[0].substring(0, i);
            }
        }
        i++;
    }
    
    return strs[0].substring(0, i);
};

File: Longest Common Prefix - Leetcode 14/Longest Common Prefix - Leetcode 14.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        
        int minLength = Integer.MAX_VALUE;
        for (String s : strs) {
            if (s.length() < minLength) {
                minLength = s.length();
            }
        }
        
        int i = 0;
        while (i < minLength) {
            for (String s : strs) {
                if (s.charAt(i) != strs[0].charAt(i)) {
                    return strs[0].substring(0, i);
                }
            }
            i++;
        }
        
        return strs[0].substring(0, i);
    }
}
